torch/fx/experimental/symbolic_shapes.py

```diff
@@ -1530,12 +1593,21 @@ def _verify(expr, potential_expr):
             for i, ss in enumerate(t.size()):
                 property_source = TensorPropertySource(source, TensorProperty.SIZE, i)
                 track_symint(property_source, ss)
-                if _is_dim_dynamic(t, i):
-                    # If this dim is marked dynamic, we need to do a test on it, to ensure that it has not bee
-                    # constrained to an integer.
-                    if _is_int(ss):
-                        raise RuntimeError(f"Attempting to constrain dim {i} for {source}, which violates user's mark_dynamic")
-                    dynamic_sources.append(property_source)
+                if dynamic_ranges:
+                    dyn_range = dynamic_ranges[pos]
+                    if dyn_range and i in dyn_range:
+                        # If this dim is marked dynamic, we need to do a test on it, to ensure that it has not bee
+                        # constrained to an integer.
+                        if _is_int(ss):
+                            raise RuntimeError(f"Attempting to constrain dimension "
```

Comment:
UX request: It would be sweet if all the verification errors were collected separately and raised together: that way, whether to raise or not could be guarded by the strict flag, but also, the errors could be structured and consumed together. Otherwise users will end up seeing one error at a time, which is not great. Also I want to build some tooling that would consume the errors in their structured form and potentially suggest constraints based off them.

Reply:
Feel free to add this mode as a stacked PR on top of this. I do not have strong feelings about the UX.

Reply:
sg